airbyte-webapp/src/components/StreamTestingPanel/LogsDisplay.tsx

```diff
@@ -1,28 +1,33 @@
+import { useMemo } from "react";
 import { FormattedMessage } from "react-intl";
 
 import { Text } from "components/ui/Text";
 
 import { StreamReadLogsItem } from "core/request/ConnectorBuilderClient";
 
 import styles from "./LogsDisplay.module.scss";
+import { formatJson } from "./utils";
 
 interface LogsDisplayProps {
   logs: StreamReadLogsItem[];
+  onTitleClick: () => void;
 }
 
-export const LogsDisplay: React.FC<LogsDisplayProps> = ({ logs }) => {
+export const LogsDisplay: React.FC<LogsDisplayProps> = ({ logs, onTitleClick }) => {
+  const formattedLogs = useMemo(() => formatJson(logs), [logs]);
+
   return (
     <div className={styles.container}>
-      <div className={styles.header}>
+      <button className={styles.header} onClick={onTitleClick}>
         <Text size="sm" bold>
           <FormattedMessage id="connectorBuilder.connectorLogs" />
         </Text>
         <Text className={styles.numLogsDisplay} size="xs" bold>
           {logs.length}
         </Text>
```

Comment:
We have a `NumberBadge` component that should ideally be used here. (I know not directly relevant to this PR, but I just noticed it). We can extend that for more colors if needed.

Reply:
I was not aware of that, thanks!